models/index.ts

```diff
@@ -248,6 +277,16 @@ export interface summarizeResponse {
   meta?: metaResponse;
 }
 
+export interface rerankResponse {
+  id: string;
+  results: {
+    document?: { text: string };
+    index: number;
+    relevance_score: number;
+  }[];
```

Comment:
Also, just a nit but I think the results could be extracted to a separate interface and we can declare the type as `results: rerankResults[]`